main_test.go

```diff
@@ -289,10 +292,12 @@ func (o SomeStruct2) DeepCopy() SomeStruct2 {
 		cp.mapStruct = make(map[string]SomeStruct, len(o.mapStruct))
 		for k2, v2 := range o.mapStruct {
 			var cp_mapStruct_v2 SomeStruct
+			cp_mapStruct_v2 = v2
```

Comment:
This looks like the fix we want to achieve, because we're only working with the mapSlice field afterwards. The rest seem redundant however, since the values are only copied if they are not nil, and for one case, its `DeepCopy` method is being invoked.